### PR TITLE
fix: allow fallback to fastify@2.22 route config

### LINKED ISSUE #411: Why @fastify-static@6.11.1 bind fastify@4.23.0 rather than 4.x
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Fastify version

4.22.2

### Plugin version

6.11.1

### Node.js version

18

### Operating system

macOS

### Operating system version (i.e. 20.04, 11.3, 10)

12.6.5

### Description

FastifyError: fastify-plugin: @fastify/static - expected '^4.23.0' fastify version, '4.22.2' is installed

https://github.com/fastify/fastify-static/blob/2471a4c2413f7958b0045073e70d81b1bd85b57d/index.js#L554-L556

Why is the only fastify version specified in version 6.11.1 4.23.0 instead of 4.x, which prevents me from running it compatibly.

### Steps to Reproduce

install @fastify/static@6.11.1 with fastify@4.22.2

### Expected Behavior

@fastify/static@6.11.1 can be used in fastify@4.x

### PR DESCRIPTION
**Pull request by climba03003**
Closes #411

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -396,7 +396,9 @@ async function fastifyStatic (fastify, opts) {
   }
 
   function serveFileHandler (req, reply) {
-    const routeConfig = req.routeOptions.config
+    // TODO: remove the fallback branch when bump major
+    /* istanbul ignore next */
+    const routeConfig = (req.routeOptions && req.routeConfig.config) || req.routeConfig
```
Comment (gurgunday) [inline index.js:401]:
```suggestion
    const routeConfig = req.routeOptions?.config || req.routeConfig
```

**Review by Uzlopak — APPROVED** [review]
LGTM, decide if you want to apply the suggestion of @gurgunday

**Review by gurgunday — CHANGES_REQUESTED** [review]
I think it should be `req.routeOptions.config` instead of `req.routeConfig.config`

**Comment by Uzlopak** [comment]
Very good catch!

**Comment by Uzlopak** [comment]
Ok, tested it locally thoroughly, by using fastify 4.20.0 and it works. updated to latest, works and does not emit warnings. modified that line to use a different key with latest fastify, emitted warnings.

**Comment by Uzlopak** [comment]
@gurgunday 
PTAL

**Review by gurgunday — APPROVED** [review]

**Review by mcollina — APPROVED** [review]
lgtm

**Review by Eomm — APPROVED** [review]

**Pull request merged**